[Tan Yuan Zheng] iP

YZTangent opened the pull request:
Dave 2 reporting for duty.

> Very cool bot. 10/10 – Anon user (not me in a wig)

>Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

MeowMeow will make sure that you never have to remember anything ever again!!! It's 

* text-based
* easy to learn
* ~FAST~ of mediocre speed to use
* is probably a weeb idk

All you need to do is:

1. download it from [here](https://github.com/YZTangent/ip/releases/download/A-Jar/Dave2.jar).
2. double-click it. 
3. add your tasks. 
4. let it manage your tasks for you 💀

And it's FREE!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

YZTangent committed:
Level-1

YZTangent committed:
Level 2

YZTangent committed:
Commit for Level 3 and Level 4.

YZTangent committed:
Added Level 5.

YZTangent committed:
Added improvements to Level 5.

YZTangent committed:
Added A-TextUiTesting.

YZTangent committed:
Updated test cases for text ui test.

YZTangent committed:
Added Level 7.

YZTangent committed:
Added Level 7.

YZTangent committed:
Updated .gitignore to ignore safe files.

YZTangent committed:
Merge branch 'branch-Level-7'

YZTangent committed:
Added Level 8.

YZTangent committed:
Merge branch 'branch-Level-8'

YZTangent committed:
Refactored code to use more OOP in line with A--MoreOOP.

YZTangent committed:
Added A-Packages.

YZTangent committed:
Added A-Gradle.

YZTangent committed:
Updated build parameters for Gradle.

YZTangent committed:
Added JavaDocs for A-JavaDocs.

YZTangent committed:
Added Level 9.

YZTangent committed:
Added A-CodingStandard.

YZTangent committed:
Merge branch 'branch-Level-9'

YZTangent committed:
Merge branch 'branch-A-CodingStandard'

ryanguai commented on src/main/java/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,29 @@
+public class AddDeadlineCommand implements Command {
+
+    private TaskList tasks;
+    private Deadlines task;
```
Is there any reason why "Deadlines" is written as plural here?

ryanguai commented on src/main/java/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    public AddDeadlineCommand(TaskList tasks, String input) throws DaveException {
+        if (input.equals("")) {
+            throw new DaveException("( ; ω ; ) Oh nyo!!! The description of an event cannot be empty!");
+        }
+        String[] args = input.split("/by ");
+        if (args.length > 2) {
+            throw new DaveException("( ; ω ; ) Oh nyo!!! Too many deadlines, Dave's brain is fried!");
+        } else if (args.length < 2) {
+            throw new DaveException("( ; ω ; ) Oh nyo!!! Please provide the time of the deadline!");
+        }
```
If you want, you can consider writing individual Exceptions for each scenario, rather than using a catch-all exception as such.

ryanguai commented on src/main/java/DaveException.java:
```diff
@@ -0,0 +1,13 @@
+public class DaveException extends Exception {
+
+    private String error;
+
+    public DaveException(String error) {
+        this.error = error;
+    }
+
+    @Override
+    public String toString() {
+        return this.error;
+    }
+}
```
You may want to have a package to house all of your exceptions. I feel use of packages would make the organisation easier to understand.

ryanguai commented on src/main/java/RemoveTaskCommand.java:
```diff
@@ -0,0 +1,34 @@
+
+    @Override
+    public String execute() throws DaveException {
+        try{
```
Minor point, missing space after "try".

ryanguai commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+         return this.done;
+     }
+
+     public String markdone() {
```
Remember to use camelCase for methods. Same for "unmarkdone" below.

ryanguai commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,28 @@
+
+     private final String task;
+
+     private boolean done = false;
```
Perhaps you can rename this to isDone? To sound more like a boolean.

ryanguai commented on src/main/java/Dave2.java:
```diff
@@ -0,0 +1,78 @@
+        }
+        try {
+            switch (command) {
+                case "bye":
+                    print("Bye. Hope to see you again soon!");
+                    running = false;
+                    break;
+                case "list":
+                    print(tasks);
+                    break;
+                case "mark":
+                    print(new MarkDoneCommand(Dave2.tasks, args).execute());
+                    break;
+                case "unmark":
+                    print(new UnmarkDoneCommand(Dave2.tasks, args).execute());
+                    break;
+                case "todo":
+                    print(new AddTodoCommand(Dave2.tasks, args).execute());
+                    break;
+                case "deadline":
+                    print(new AddDeadlineCommand(Dave2.tasks, args).execute());
+                    break;
+                case "event":
+                    print(new AddEventCommand(Dave2.tasks, args).execute());
+                    break;
+                case "remove":
+                    print(new RemoveTaskCommand(Dave2.tasks, args).execute());
```
Since these are constants, you may consider using enums if you wish.

ryanguai commented on src/main/java/Dave2.java:
```diff
@@ -0,0 +1,78 @@
+
+    private static boolean running = true;
+
+    private static final String line = "____________________________________________________________\n";
```
If this is a constant, you can write in caps.

YZTangent committed:
Refactored text output into textui class, and logic switching into the
parser class.

YZTangent committed:
Added basic echo functionality to Dave 2's GUI.

YZTangent committed:
Added profile pictures for user and Dave 2.

YZTangent committed:
Added complete functionalities to the GUI such that the user is able to
interact with Dave 2 through the GUI only.

YZTangent committed:
Used FXML in implemeting Dave 2's GUI.

YZTangent committed:
Check code quality for A-CodeQuality

YZTangent committed:
Add asserts to key points for A-Assertions

YZTangent committed:
Add archive functionality to Dave2

Archive functionality is a useful feature for users so that
they can save tasks that they do not want in their current list.

Archival function has common behaviours with the save file
handling function.

Creating an archival feature using the save file
handling abstraction allows us to avoid repeating ourselves
(hence folliwng DRY).

YZTangent committed:
Add archive directory to .gitignore

YZTangent committed:
Add A-Personality

YZTangent committed:
Add resize functionality and more visual improvements to GUI

The GUI is unable to be resized and the colour palette is quite bland,
and does not fit the personality of the bot.

Adding more the resize functionality will make the bot more versatile
and adaptable to different display sizes. Adding a blue/pink colour
palette as well as rounded dialog box corners creates a more friendly
user experience.

YZTangent committed:
Add User Guide for Kohaku bot

YZTangent committed:
Merge branch 'master' into branch-A-Assertions

YZTangent committed:
Merge pull request #2 from YZTangent/branch-A-Assertions

Add asserts to key points for A-Assertions

YZTangent committed:
Merge branch 'master' into branch-A-CodeQuality

YZTangent committed:
Merge pull request #3 from YZTangent/branch-A-CodeQuality

Check code quality for A-CodeQuality

YZTangent committed:
Fix minor style issue as well as naming bugs

YZTangent committed:
Refactor project directory and add test cases

Refactor project directory to allow for the adding of test cases.
Adding test cases for Parser, MarkDoneCommand and Tasklist to test
more complicated pieces of logic.

YZTangent committed:
Edit build parameters for A-Release

YZTangent committed:
Fix for Ui image in User guide
